linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/NameCompressor.scala

```diff
@@ -107,24 +119,50 @@ private[emitter] object NameCompressor {
   private val BasePropertyNamesToAvoid: Set[String] =
     NameGen.ReservedJSIdentifierNames + "then"
 
-  /** Keys of this map are `(ClassName, FieldName) | MethodName`. */
-  private type EntryMap = mutable.AnyRefMap[AnyRef, PropertyNameEntry]
-
-  private sealed abstract class PropertyNameEntry {
-    var occurrences: Int = 0
-    var allocatedName: String = null
+  private sealed abstract class BaseEntry {
+    private var _occurrences: Int = 0
+    private var allocatedName: String = null
+    private var actuallyUsed: Boolean = false
```

Comment:
If we decide to keep the "usage only" check with this high amount of mutability, we might want to re-use `_occurrences` to store usage (set it to 0 when used).